controllers/comments.controller.js

```diff
@@ -10,3 +14,22 @@ exports.getCommentsByArticleID = (req, res, next) => {
    next(err);
   });
 };
+
+exports.postComment = (req, res, next) => {
+ const { article_id } = req.params;
+ const { username, body } = req.body;
+
+ fetchUsersByUsername(username)
+  .then(() => {
+   createComment(article_id, username, body)
+    .then((comment) => {
+     res.status(201).send({ comment });
+    })
+    .catch((err) => {
+     next(err);
+    });
+  })
+  .catch((err) => {
+   next(err);
```

Comment:
Here you have two asynchronous functions, and there are two ways of dealing with this to avoid _nesting_...
- `return` the `createComment` function from the first block so the promise is resolved in a second _chained_ then block
- or make use of `promise.all` but I think you were having difficulties with not getting the right error back which leads me to my third and most important point
- You don't need to have two model functions here, the `author` and `article_Id` are both _foreign keys__ on the comments table so all you need is the `createComment` function and PSQL will throw the errors for you if author or article_id are not found